### PR TITLE
feat(triton-linalg): add ci for triton-linalg

### PR DESCRIPTION
**Pull request by zhangqirui**

### CODE REVIEW

**Comment by zhangqirui** [comment]
流水分为4个步骤：clone，check_pr, build ,test

File: .github/ci_script/combine_log.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import time
+import sys
+import os
+'''
+  Get info.
```
Comment (sethbrin) [inline .github/ci_script/combine_log.py:5]: Get what info? Comments more about this file.

File: .github/ci_script/combine_log.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    output_path: the target file that you want to combine sub log with.
+    list_path: the list of sub log name. When it is updated, the correspondding file will be add to output tail.
+    list_dir_path: the dir path where sub logs stored.
+    status_path: the path of status file. When status file is written to "success" or "fail", exit script.
+'''
```
Comment (sethbrin) [inline .github/ci_script/combine_log.py:10]: use argparse, you may use chatgpt to help rewrite this scripts.

File: .github/ci_script/file_guard.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import time
+import sys
+import os
+guard_status_file = sys.argv[1]
```
Comment (sethbrin) [inline .github/ci_script/file_guard.py:4]: ditto

File: .github/ci_script/triton-linalg-ci_script.sh [hunk 4]
```diff
@@ -0,0 +1,90 @@
+# /bin/bash
+# get PR id
```
Comment (sethbrin) [inline .github/ci_script/triton-linalg-ci_script.sh:2]: nit: Get PR id

File: .github/ci_script/triton-linalg-ci_script.sh [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+# gen name of this ci
+request_name="${repo_name}_${pr_id}_${currentTimeStamp}_${card_type}.rqt"
+
+# gen file and dir for this request
+request_root="$repo_root/$request_name/"
+sub_logs_path="$request_root/sub_logs/"
+
+
+# echo "${repo_root}"
```
Comment (sethbrin) [inline .github/ci_script/triton-linalg-ci_script.sh:35]: remove useless comments

File: .github/ci_script/triton-linalg-ci_script.sh [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+if [ ! -d $sub_logs_path ];then
+    mkdir $sub_logs_path
+fi
+
+echo "working" > "$request_root/status"
+chmod o+w "$request_root/status"
+
+if [ ! -f  "$request_root/log" ];then
+	touch "$request_root/log"
```
Comment (sethbrin) [inline .github/ci_script/triton-linalg-ci_script.sh:51]: code style, indent

File: .github/ci_script/triton-linalg-ci_script.sh [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+chmod o+w "$request_root/log"
+
+if [ ! -f "$request_root/log_list" ];then
+    touch "$request_root/log_list"
+fi
+
+chmod o+w "$request_root/log_list"
+
+# gen request file.
+#echo "${repo_name},${pr_id},${currentTimeStamp},${card_type}" > "$requests_path/${request_name}"
```
Comment (sethbrin) [inline .github/ci_script/triton-linalg-ci_script.sh:63]: ditto

File: .github/ci_script/triton-linalg-ci_script.sh [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+echo "repo:${repo_name}" > "$requests_path/${request_name}"
+echo "pr_id:${pr_id}" >> "$requests_path/${request_name}"
+echo "timestamp:${currentTimeStamp}" >> "$requests_path/${request_name}"
+
+# change dir group for server and client, or when server/client try to delete request, ftp may raise error.
+
+#chgrp -R ftpuser $request_root
```
Comment (sethbrin) [inline .github/ci_script/triton-linalg-ci_script.sh:71]: ditto

File: .github/ci_script/triton-linalg-ci_script.sh [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+#chgrp -R ftpuser $requests_path
+
+# start script
+python3 .github/ci_script/file_guard.py "$request_root/status" "$request_root/log" &
+python3 .github/ci_script/combine_log.py "$request_root/log" "$request_root/log_list" "$request_root/sub_logs" "$request_root/status" &
+
+wait
+
+# status=$(cat ${request_root}/status)
```
Comment (sethbrin) [inline .github/ci_script/triton-linalg-ci_script.sh:80]: ditto

File: tools/ci/daily/triton-linalg_daliy.pipeline [hunk 10]
```diff
@@ -0,0 +1,118 @@
+library "cambricon-pipe-lib@master"
+cnpipe {
+    task('clone') {
+	stage 'clone'
+        runOnCloud false
+        node {
+             label "compiler_test_10.100.146.123"
```
Comment (sethbrin) [inline tools/ci/daily/triton-linalg_daliy.pipeline:7]: why use this server

File: tools/scripts/lint_check/IWYU.md [hunk 11]
```diff
@@ -0,0 +1,9 @@
+使用IWYU报错后有如下一些原则：
```
Comment (sethbrin) [inline tools/scripts/lint_check/IWYU.md:1]: translate to English or just remove this file

File: tools/scripts/lint_check/format_checker.py [hunk 12]
```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+#
+#===- format_checker.py - Format check&gen-------------------*- python3 -*--===#
+#
+# Copyright (C) [2022-2025] by Cambricon
```
Comment (sethbrin) [inline tools/scripts/lint_check/format_checker.py:5]: remove this format_checker.py

File: .github/ci_script/combine_log.py (hunk 1 above)
Comment (zhangqirui) [reply .github/ci_script/combine_log.py]: Done

File: .github/ci_script/combine_log.py (hunk 2 above)
Comment (zhangqirui) [reply .github/ci_script/combine_log.py:11]: Done

File: .github/ci_script/file_guard.py (hunk 3 above)
Comment (zhangqirui) [reply .github/ci_script/file_guard.py]: Done

File: .github/ci_script/triton-linalg-ci_script.sh (hunk 4 above)
Comment (zhangqirui) [reply .github/ci_script/triton-linalg-ci_script.sh]: Done

File: .github/ci_script/triton-linalg-ci_script.sh (hunk 5 above)
Comment (zhangqirui) [reply .github/ci_script/triton-linalg-ci_script.sh]: Done

File: .github/ci_script/triton-linalg-ci_script.sh (hunk 8 above)
Comment (zhangqirui) [reply .github/ci_script/triton-linalg-ci_script.sh]: Done

File: .github/ci_script/triton-linalg-ci_script.sh (hunk 9 above)
Comment (zhangqirui) [reply .github/ci_script/triton-linalg-ci_script.sh]: Done

File: tools/ci/daily/triton-linalg_daliy.pipeline (hunk 10 above)
Comment (zhangqirui) [reply tools/ci/daily/triton-linalg_daliy.pipeline]: Modified to cloud platform server

File: tools/scripts/lint_check/IWYU.md (hunk 11 above)
Comment (zhangqirui) [reply tools/scripts/lint_check/IWYU.md]: Done

File: tools/scripts/lint_check/format_checker.py (hunk 12 above)
Comment (zhangqirui) [reply tools/scripts/lint_check/format_checker.py]: Done

File: .github/ci_script/triton-linalg-ci_script.sh (hunk 6 above)
Comment (zhangqirui) [reply .github/ci_script/triton-linalg-ci_script.sh]: done

**Pull request merged**